### PR TITLE
delete asset exchange

### PR DESCRIPTION
**Pull request by ducquangkstn**

### CODE REVIEW

File: v3/storage/postgres/asset_exchange.go [hunk 1]
```diff
@@ -51,3 +65,25 @@ func (s *Storage) GetAssetExchangeBySymbol(exchangeID uint64, symbol string) (co
 		return result, fmt.Errorf("failed to get asset from database symbol=%s err=%s", symbol, err.Error())
 	}
 }
+
+func (s *Storage) deleteAssetExchange(tx *sqlx.Tx, assetExchangeID uint64) error {
+	var returnedID uint64
+
+	row := tx.Stmt(s.stmts.deleteAssetExchange.Stmt).QueryRow(assetExchangeID)
```
Comment (secmask) [inline v3/storage/postgres/asset_exchange.go]: why not `err = tx.Stmtx(s.stmts.deleteAssetExchange.Stmt).Get(&assetExchangeID)`

File: v3/storage/postgres/setting_change.go [hunk 2]
```diff
@@ -244,7 +244,18 @@ func (s *Storage) applyChange(tx *sqlx.Tx, i int, entry common.SettingChangeEntr
...
-		// TODO: implement delete
+		a, ok := entry.Data.(*common.DeleteAssetExchangeEntry)
+		if !ok {
+			msg := fmt.Sprintf("bad cast at %d to %s\n", i, common.ChangeTypeDeleteAssetExchange)
+			log.Println(msg)
+			return errors.Wrap(err, msg)
+		}
+		err = s.deleteAssetExchange(tx, a.AssetExchangeID)
+		if err != nil {
+			msg := fmt.Sprintf("update exchange %d, err=%v\n", i, err)
```
Comment (secmask) [inline v3/storage/postgres/setting_change.go]: `update exchange`  -> `delete asset exchange`

Comment (ducquangkstn) [reply v3/storage/postgres/setting_change.go]: fix as request

File: v3/storage/postgres/asset_exchange.go (hunk 1 above)
Comment (ducquangkstn) [reply v3/storage/postgres/asset_exchange.go]: fix as request

File: v3/common/errors.go [hunk 3]
```diff
@@ -46,4 +46,6 @@ var (
 	ErrTradingByAssetIDInvalid = errors.New("asset_id is either base or quote of trading_pair")
 	// ErrInvalidAddress is when an address is not an ethereum address.
 	ErrInvalidAddress = errors.New("address is not an ethereum address")
+	// ErrAssetExchangeDeleteViolation is returned when there are trading pair relating to deleted asset exchange
+	ErrAssetExchangeDeleteViolation = errors.New("all trading pair relating to asset exchange should be deleted")
```
[the remark this answers is not in the document]
Comment (ducquangkstn) [reply v3/common/errors.go]: fix as request

**Pull request merged**